client/client.go

```diff
@@ -67,63 +63,43 @@ func (c *Client) CompletedObjectives() <-chan protocols.ObjectiveId {
 
 // CreateVirtualChannel creates a virtual channel with the counterParty using ledger channels with the intermediary.
 func (c *Client) CreateVirtualChannel(counterParty types.Address, intermediary types.Address, appDefinition types.Address, appData types.Bytes, outcome outcome.Exit, challengeDuration *types.Uint256) protocols.ObjectiveId {
-	right, ok := (*c.store).GetTwoPartyLedger(*c.Address, intermediary)
-
-	if !ok {
-		// TODO: We need to implement proper API error handling
-		panic(fmt.Sprintf("Could not find ledger channel for participants %v,%v", *c.Address, intermediary))
+	objectiveRequest := virtualfund.ObjectiveRequest{
+		MyAddress:         *c.Address,
+		CounterParty:      counterParty,
+		Intermediary:      intermediary,
+		AppDefinition:     appDefinition,
+		AppData:           appData,
+		Outcome:           outcome,
+		ChallengeDuration: challengeDuration,
+		Nonce:             rand.Int63(), // TODO: Proper nonce management!
 	}
-
-	var left *channel.TwoPartyLedger
-
-	// Convert the API call into an internal event.
-	objective, _ := virtualfund.NewObjective(true,
-		state.State{
-			ChainId:           big.NewInt(0), // TODO
-			Participants:      []types.Address{*c.Address, intermediary, counterParty},
-			ChannelNonce:      big.NewInt(0), // TODO -- how do we get a fresh nonce safely without race conditions? Could we conisder a random nonce?
-			AppDefinition:     appDefinition,
-			ChallengeDuration: challengeDuration,
-			AppData:           appData,
-			Outcome:           outcome,
-			TurnNum:           0,
-			IsFinal:           false,
-		},
-		*c.Address,
-		left, right)
-
 	apiEvent := engine.APIEvent{
-		ObjectiveToSpawn: &objective,
+		ObjectiveToSpawn: objectiveRequest,
```

Comment:
Should `ObjectiveToSpawn` be renamed, since it is no longer an objective to be spawned?

Reply:
We could rename it to `ObjectiveSpawnRequests`?

Reply:
I think that's up to you. I'm just pointing out that `ObjectiveToSpawn` used to contain an objective, which was supposed to be spawned. Now, it doesn't, and so the name seems stale?

Reply:
I think I'm going to leave this as is for now as it follows the existing pattern of `ObjectiveToApprove` and `ObjectiveToReject`.